### ISSUE TITLE
mac tests failing with "Too many open files in system" error

### ISSUE THREAD

**Issue by markdroth** [grpc/grpc#16985]
I've seen this a number of times recently.  There are two examples here:

https://source.cloud.google.com/results/invocations/7708f38a-ffbf-4587-ac60-ddb4fce4be02/targets/github%2Fgrpc%2Fcpp_macos_opt_native/tests

I assume there's some resource limitation on our mac workers.  Can someone take a look?  Thanks!

**Comment by jtattermusch** [comment]
The first build where this is failing is:
https://source.cloud.google.com/results/invocations/ce056eff-7ad2-49b8-b70d-7853525987e8 from 2018-10-30 22:38:50 UTC. Since then it's failing for almost all builds.

**Comment by jtattermusch** [comment]
This is still happening.
I tried to bisect but wasn't able to pinpoint the cause.
Our macos prepare scripts are actually setting `ulimit -n` so changes in macos worker configuration shouldn't really affect us.
https://github.com/grpc/grpc/blob/6d0add97736b781acbf5a155229e14cb6200253e/tools/internal_ci/helper_scripts/prepare_build_macos_rc#L23

Current values used by macos workers (AFAIK there has been no change for a while):
```
+ source tools/internal_ci/helper_scripts/prepare_build_macos_rc
++ sudo launchctl limit maxfiles unlimited unlimited
++ launchctl limit maxfiles
	maxfiles    10240          10240          
++ ulimit -n 10000
++ ulimit -a
core file size          (blocks, -c) 0
data seg size           (kbytes, -d) unlimited
file size               (blocks, -f) unlimited
max locked memory       (kbytes, -l) unlimited
max memory size         (kbytes, -m) unlimited
open files                      (-n) 10000
pipe size            (512 bytes, -p) 1
stack size              (kbytes, -s) 8192
cpu time               (seconds, -t) unlimited
max user processes              (-u) 2500
virtual memory          (kbytes, -v) unlimited
```

**Comment by jtattermusch** [comment]
More data: 
- This is happening mostly for `bins/opt/client_channel_stress_test` and `bins/opt/grpclb_end2end_test` tests. 
- It only happens for `opt`, doesn't happen in `dbg`.

**Comment by jtattermusch** [comment]
Seems fixed by #17187
